pkg/registry/core/service/storage/rest.go

```diff
@@ -390,13 +371,12 @@ func (rs *REST) healthCheckNodePortUpdate(oldService, service *api.Service, node
 		klog.Infof("Transition to non LoadBalancer type service or LoadBalancer type service with ExternalTrafficPolicy=Global")
 		klog.V(4).Infof("Releasing healthCheckNodePort: %d", oldHealthCheckNodePort)
 		nodePortOp.ReleaseDeferred(int(oldHealthCheckNodePort))
-		// Clear the HealthCheckNodePort field.
-		service.Spec.HealthCheckNodePort = 0
```

Comment:
is ok to remove it completely or should we have something on strategy clearing the healthchecknodeport?

Reply:
it is already cleared in `dropTypeDependentFields` , we can resolve this comment